daffodil-io/src/main/scala/org/apache/daffodil/io/InputSource.scala

```diff
@@ -142,15 +173,21 @@ abstract class InputSource {
  *
  * @param inputStream the java.io.Inputstream to read data from
  * @param bucketSize the size of each individual bucket
+ * @param maxCacheSizeInBytes the maximum number of buckets that can be created before beginning to remove old ones
  */ 
-class BucketingInputSource(inputStream: java.io.InputStream, bucketSize: Int = 1 << 13)
+class BucketingInputSource(
+  inputStream: java.io.InputStream,
+  bucketSize: Int = 1 << 13,
+  maxCacheSizeInBytes: Int = 512000)
```

Comment:
This is only 0.5 MB. Any concerns with this? While it should be large enough for any backtracking case, we certainly have files that are significantly larger than this. So this could break backwards compatbilitiy if they actually relied on the ability to backtrack that far. And there is currently no way for a user to change it if it is too small.

This seems like a reasonable default only if there is a way for the user to change it. Maybe that can be done as another bug for this release? I know another comment mentioned some ways to allow configuring this via tunables.